lib/node.rb

```diff
@@ -1,4 +1,5 @@
 class Node
+  include Comparable
```

Comment:
I included the `Comparable` interface here just to save myself from having to unwrap the nodes each time I want to do a comparison.